### PR TITLE
scripts: Configuration channel application for nRF52 Desktop

### PR DESCRIPTION
**Pull request by Qbicz**
Application works with nRF52 Desktop firmware in order to control the
firmware parameters at runtime. It uses HID feature reports to
communicate with the device.

In initial version, controlling the mouse sensor resolution (CPI) is
supported.

Jira: DESK-301

Signed-off-by: Filip Kubicz <filip.kubicz@nordicsemi.no>

### CODE REVIEW

**Comment by Qbicz** [comment]
I marked it as DNM because I have a few questions:
* @carlescufi there is a `requirements` file in the top of `scripts` directory. Wouldn't it conflict if we have also `requirements.txt` in `scripts/hid_configurator`?
* I want to attach a DLL which is 47kB big. Can it live in the repository, or should be downloadable from external location?

**Comment by Qbicz** [comment]
Can you review @pdunaj @MarekPieta?

**Comment by carlescufi** [comment]
>@carlescufi there is a requirements file in the top of scripts directory. Wouldn't it conflict if we have also requirements.txt in scripts/hid_configurator?

Just include that single line in the common `scrtips/requirement.txt`, it won't hurt anyone to have it even if they don't use it.

>I want to attach a DLL which is 47kB big. Can it live in the repository, or should be downloadable from external location?

Why not have the DLL in the python repo which is installed by `pip`?

**Comment by Qbicz** [comment]
@carlescufi I moved the DLL, it is now installed with the repo via `pip`.

For the requirements file, I prefer to keep it separate. The top-level `requirements.txt` installs sphinx and tools for documentation (pretty large package), which may not be needed by people using the tool.

File: scripts/hid_configurator/README.md [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+```
+python configurator.py --cpi 12000
+```
+
+# Installation
+To use configurator program, you need:
+* HIDAPI library - on Windows this is provided by attached `hidapi.dll`
+* pyhidapi Python wrapper - run
+```
+pip install -r requirements.txt
```
Comment (carlescufi) [inline scripts/hid_configurator/README.md]: Use `.rst` and not `.md`.

Also:
```
# On Linux:
pip3 install --user -r requirements.txt
# On macOS and Windows:
pip3 install --user -r requirements.txt
```

File: scripts/hid_configurator/README.md [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+Right now, it can be used to change following parameters:
+* mouse optical sensor resolution (CPI)
+
+Supported transports:
+* USB
+
+# Usage
+```
+python configurator.py --cpi 12000
```
Comment (carlescufi) [inline scripts/hid_configurator/README.md]: You probably want `python3` on Linux/macOS and `py -3` on Windows

**Comment by Qbicz** [comment]
Switched to `.rst`, added Windows and Linux instructions for Python 3. Rebased.

File: scripts/hid_configurator/README.rst [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+On Linux:
+
+	python3 configurator.py --cpi 12000
+
+Installation
+************
+To setup dependencies for the configurator program, run:
+On Windows:
+
+	py -3 -m pip install --user -r requirements.txt
```
Comment (carlescufi) [inline scripts/hid_configurator/README.rst]: you can actually use `pip3` on Windows but that's fine.

**Comment by Qbicz** [comment]
Hi @pizi-nordic, can you merge?

**Pull request merged**